frontend/src/LineGraph.res

```diff
@@ -477,46 +525,72 @@ let make = React.memo((
     Js.Float.isNaN(x) ? "?" : Js.Float.toPrecisionWithPrecision(~digits, x)
   }
 
-  let lastValues =
-    dataSet->Belt.Array.map(x =>
-      x->BeltHelpers.Array.lastExn->DataRow.toValue->floatToStringHandleNaN
+  let handleLegendClick = (event: ReactEvent.Mouse.t) => {
+    let target = ReactEvent.Mouse.target(event)
+    let label = searchDataLabel(target)
+    let index =
+      labels
+      ->Belt.Option.getWithDefault([])
+      ->Belt.Array.getIndexBy(l => l == label)
+      ->Belt.Option.getWithDefault(-1)
+    setLinesState(_ =>
+      linesState->Belt.Array.mapWithIndex((idx, state) => idx == index ? !state : state)
     )
-  let isOverlayed = lastValues->Belt.Array.length > 1
+  }
+  let labels = labels->Belt.Option.getWithDefault([])
+  let lastValues = dataSet->Belt.Array.map(x => x->BeltHelpers.Array.lastExn->DataRow.toValue)
+  let lastValues = Belt.Array.zip(labels, lastValues)->Belt.Map.String.fromArray
+  let linesStateByLabel = Belt.Array.zip(labels, linesState)->Belt.Map.String.fromArray
+  let legendColorsByLabel = Belt.Array.zip(labels, legendColors)->Belt.Map.String.fromArray
+  let isOverlayed = labels->Belt.Array.length > 1
   let right = isOverlayed
     ? <Row
         spacing=#between
         sx=[
-          Sx.d.inlineFlex,
-          Sx.flex.col,
           Sx.items.start,
           Sx.flex.wrap,
           Sx.unsafe("width", "min-content"),
+          Sx.unsafe("display", "grid"),
+          Sx.unsafe("gridTemplateColumns", "auto auto auto auto auto auto"),
+          Sx.unsafe("gap", "8px"),
         ]>
-        {originalLabels
-        ->Belt.Option.getWithDefault([])
+        {labels
+        ->Belt.SortArray.stableSortBy((labelA, labelB) =>
+          compare(
+            lastValues->Belt.Map.String.getWithDefault(labelB, Js.Float._NaN),
+            lastValues->Belt.Map.String.getWithDefault(labelA, Js.Float._NaN),
+          )
+        )
         ->Belt.Array.mapWithIndex((idx, label) => {
           let hex =
-            legendColors
-            ->Belt.Array.get(idx)
-            ->Belt.Option.getWithDefault("#000000")
+            legendColorsByLabel
+            ->Belt.Map.String.getWithDefault(label, "#000000")
             ->Js.String2.substr(~from=1)
+          let active = linesStateByLabel->Belt.Map.String.getWithDefault(label, true)
           <div
             key={idx->Belt.Int.toString}
+            data={label}
+            onClick={handleLegendClick}
```

Comment:
Would it make sense to pass `label` to `handleLegendClick` with a partial application? If I understand correctly, the raw javascript `searchDataLabel` is searching the DOM for the above `data={label}`, but then we would'nt need to do that if the handler already had `label`?

Reply:
That's right. Thanks for catching this. :)